outbox-core/src/main/java/io/praesidio/outbox/MessageRelayProvider.java

```diff
@@ -0,0 +1,8 @@
+package io.praesidio.outbox;
+
+import io.praesidio.outbox.values.MessageType;
+
+public interface MessageRelayProvider {
+    void relay(SendMessageCommand sendMessageCommand);
```

Comment:
Can we use here `Message` as a parameter?

Then method `SendMessageCommand convert(Message message)` from `MessageSerializer` won't be needed.

Reply:
I don't think we can. `MessageRelayProvider` needs the de-serialized content and metadata to send the message, and `Message` contains both of them in a serialized form. Maybe we should rename the two fields in `Message` to `serializedContent` and `serializedMetadata` to avoid confusion?

I think doing this the way you suggested would require the de-serialization logic to be moved from `MessageSerializer` to `MessageRelayProvider`. In my opinion this would be counter intuitive and a bit messy. Plus it will be easier to break serialization/de-serialization logic because it would be stored in two different places.

What do you think? Did I miss something here?

Reply:
For me it was a litte bit confusing that we used again the same class (`command`) - but IMO we will 'polish' API with first implementations of `MessageSerializer` and `MessageRelayProvider` :)